### ISSUE TITLE
Specify Slack instance

### ISSUE THREAD

**Issue by loleg** [singingwolfboy/flask-dance#316]
When clicking the login button provided in #40 and the user is logged into multiple Slack spaces, but NOT into the space for which you have configured Flask Dance, then you get an error like this from Slack:

```
Something went wrong when authorizing this app.
invalid_team_for_non_distributed_app
```

If you adjust the URL, putting your target Slack in the subdomain, then a login screen is presented, and the dance is completed fine.

I believe we need to be able to specify a subdomain in the `authorization_url` on the Slack side for the OAuth request, and unless I'm mistaken, this is [hard coded](https://github.com/singingwolfboy/flask-dance/blob/main/flask_dance/contrib/slack.py#L64), and not one of the currently available options.

**Comment by daenney** [comment]
Feel free to send a PR! You can replicate the same logic with being able to extract `authorization_url` using `from_config` like we do for `client_id` and `client_secret`.

**Comment by daenney** [comment]
This should now be fixed with #331

**Comment by loleg** [comment]
Thank you so much @daenney @firejake308 @singingwolfboy ! You just made my day :man_dancing:
